### PR TITLE
grib-api: add support for PNG

### PR DESCRIPTION
**Pull request by jfsimoneau**
Added --with-libpng to support GRIB2 file encoded as PNG, like MRMS.

### CODE REVIEW

File: Formula/grib-api.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ class GribApi < Formula
   homepage "https://software.ecmwf.int/wiki/display/GRIB/Home"
   url "https://software.ecmwf.int/wiki/download/attachments/3473437/grib_api-1.14.7-Source.tar.gz"
   sha256 "a42b9b0bd6bd2364897c13feafd09adba6a52e05866db61d2d7ab5ee0534f1f7"
+  revision 1
```
Comment (DomT4) [inline Formula/grib-api.rb]: Don't need a revision for adding an optional dependency.

File: Formula/grib-api.rb [hunk 2]
```diff
@@ -25,6 +27,7 @@ def install
     mkdir "build" do
       args = std_cmake_args
       args << "-DBUILD_SHARED_LIBS=OFF" if build.with? "static"
+      args << "-DPNG_PNG_INCLUDE_DIR=#{HOMEBREW_PREFIX}/include" << "-DENABLE_PNG=ON" if build.with? "libpng"
```
Comment (DomT4) [inline Formula/grib-api.rb]: Instead of `#{HOMEBREW_PREFIX}/include` should use `#{Formula["libpng"].opt_include}`

**Comment by jfsimoneau** [comment]
Thank you for the comments, I updated the Formula as requested.

File: Formula/grib-api.rb [hunk 3]
```diff
@@ -25,6 +26,7 @@ def install
     mkdir "build" do
       args = std_cmake_args
       args << "-DBUILD_SHARED_LIBS=OFF" if build.with? "static"
+      args << "-DPNG_PNG_INCLUDE_DIR=#{Formula["libpng"].opt_include}/include" << "-DENABLE_PNG=ON" if build.with? "libpng"
```
Comment (DomT4) [inline Formula/grib-api.rb]: Sorry if I wasn't clear here. You don't need the extra `/include` when calling `opt_include`.

**Comment by jfsimoneau** [comment]
Right, I missed that. I wonder why my test passed, though.

**Comment by DomT4** [comment]
Optional dependencies aren't CI tested at the present time.

**Comment by jfsimoneau** [comment]
Oh no, I mean when I tested locally. I have a GRIB2 file with PNG encoding I'm trying to read but if I build grib-api without specifying the path to libpng's include directory, CMake finds the system's libpng and the tools won't run because of a mismatch between Homebrew's libpng and the system's.

**Comment by apjanke** [comment]
Looks good now.

Merged. Thank you for your contribution to Homebrew!

**Pull request closed without merging**